File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/TransactionsPresenterModule.java
package com.fintechplatform.ui.transactions.ui;

import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.api.transactions.api.TransactionsAPI;
import com.fintechplatform.ui.transactions.db.TransactionPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TransactionsPresenterModule {

    private TransactionsContract.View view;
    private DataAccount configuration;

    public TransactionsPresenterModule(TransactionsContract.View view, DataAccount configuration) {
        this.view = view;
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    TransactionsContract.Presenter providesTransactionsPresenter(TransactionsAPI api,
                                                                 TransactionPersistanceDB transactionsPersistance) {
        return new TransactionsPresenter(view, api, configuration, transactionsPersistance);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/di/TransactionsPresenterModule.kt
package com.fintechplatform.ui.transactions.di

import com.fintechplatform.api.transactions.api.TransactionsAPI
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.transactions.db.TransactionPersistanceDB
import com.fintechplatform.ui.transactions.ui.TransactionsContract
import com.fintechplatform.ui.transactions.ui.TransactionsPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class TransactionsPresenterModule(private val view: TransactionsContract.View, private val configuration: DataAccount) {

    @Provides
    @Singleton
    internal fun providesTransactionsPresenter(api: TransactionsAPI,
                                               transactionsPersistance: TransactionPersistanceDB): TransactionsContract.Presenter {
        return TransactionsPresenter(view, api, configuration, transactionsPersistance)
    }
}
